### PR TITLE
HAAR-116 Send AuditEntry to audit api when prison is updated

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/SecurityUserContext.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.prisonregister.config
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/SecurityUserContext.kt:1]: I copied SecurityUserContext and UserContext from other services...but wondering if I could combine them.  Will do this as a separate PR

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt [hunk 2]
```diff
@@ -3,15 +3,38 @@ package uk.gov.justice.digital.hmpps.prisonregister.config
...
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction
+import org.springframework.web.reactive.function.client.ExchangeFunction
 import org.springframework.web.reactive.function.client.WebClient
+import uk.gov.justice.digital.hmpps.prisonregister.utils.UserContext
 
 @Configuration
 class WebClientConfiguration(
-  @Value("\${oauth.endpoint.url}") private val oauthRootUri: String
+  @Value("\${oauth.endpoint.url}") private val oauthRootUri: String,
+  @Value("\${audit.endpoint.url}") val auditBaseUri: String
```
Comment (neilmendum) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt:16]: This can be `private` too?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt [hunk 3]
```diff
@@ -3,15 +3,38 @@ package uk.gov.justice.digital.hmpps.prisonregister.config
...
+
```
Comment (neilmendum) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt:17]: Small one - newline in the constructor

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt]: done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt (hunk 3 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/config/WebClientConfiguration.kt]: removed

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonMaintenanceResourceIntTest.kt [hunk 4]
```diff
@@ -85,6 +93,8 @@ class PrisonMaintenanceResourceIntTest : IntegrationTest() {
         .exchange()
         .expectStatus().isOk
         .expectBody().json("updated_prison".loadJson())
+
+      verify(auditApiService).auditPrisonEvent(any())
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonMaintenanceResourceIntTest.kt]: Changed to use auditservice mock and check details

**Review by karenmillermoj — COMMENTED** [review]
Have now updated after PR review and have also moved the authorisation for auditing to the prison-register-api-client rather than the user

**Review by neilmendum — APPROVED** [review]
LGTM
